openpype/hosts/maya/api/plugin.py

```diff
@@ -404,20 +404,59 @@ def create(self, subset_name, instance_data, pre_create_data):
         # A Renderlayer is never explicitly created using the create method.
         # Instead, renderlayers from the scene are collected. Thus "create"
         # would only ever be called to say, 'hey, please refresh collect'
-        self.create_singleton_node()
+        has_singleton = bool(self._get_singleton_node())
+        if not has_singleton:
+            self.create_singleton_node()
+
+        create_mode = pre_create_data.get("create_renderlayer",
+                                          "create_first_renderlayer")
+
+        # Create the renderlayer if user chose layer creation
+        render_layer = None
```

Comment:
Hmm, that is nested... Could that be a method?
```python
        ...
        render_layer = self._get_render_layer(pre_create_data, instance_data)
        ...

    def _get_render_layer_node(self, pre_create_data, instance_data):
        create_mode = pre_create_data.get(
            "create_renderlayer", "create_first_renderlayer"
        )
        if create_mode not in {"create", "create_first_renderlayer"}:
            return None

        rs = renderSetup.instance()
        existing_layers = rs.getRenderLayers()
        if create_mode != "create" and existing_layers:
            return None

        variant = instance_data["variant"]
        for layer in existing_layers:
            if layer.name() == variant:
                raise CreatorError(
                    "Can't create multiple renderlayers of the "
                    "same variant: {}.\nThis is because the variant "
                    "defines the renderlayer name.".format(variant)
                )

        render_layer = rs.createRenderLayer(variant)
        collection = render_layer.createCollection("defaultCollection")
        collection.getSelector().setPattern('*')
        return render_layer
```

Reply:
Sure could be, yes!

Reply:
Implemented with https://github.com/ynput/OpenPype/pull/6042/commits/b8aafcbd92c199c5b53cc5693a432ea3f28c0e49